engine/timer.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <chrono>
+#include <functional>
+#include <memory>
+#include <queue>
+#include <vector>
+
+namespace pong
+{
+
+struct TimerRequest
+{
+    int mGameObjectId { -1 };
+    int mTimeoutUs { -1 };
```

Comment:
would recommend at least a named constant. If this is something that may or may not be there, I would recommend possibly looking into `std::optional`, very convenient for avoiding weird sentinels like this